src/modules/modals/ImportPreview.vue

```diff
@@ -0,0 +1,299 @@
+import ProgressIcon from 'vue-material-design-icons/ArrowRightThin.vue'
+import SelectionIcon from 'vue-material-design-icons/FormSelect.vue'
+import DatetimeIcon from 'vue-material-design-icons/ClipboardTextClockOutline.vue'
+import { ColumnTypes } from '../../shared/components/ncTable/mixins/columnHandler.js'
+import { emit } from '@nextcloud/event-bus'
+
+export default {
+	name: 'ImportPreview',
+
+	components: {
+		NcCheckboxRadioSwitch,
+		NcButton,
+		NcSelect,
+		TextLongIcon,
+		LinkIcon,
+		CounterIcon,
+		StarIcon,
+		ProgressIcon,
+		SelectionIcon,
+		DatetimeIcon,
+	},
+
+	props: {
+		previewData: {
+			type: Object,
+			default() {
+				return {
+					columns: [],
+					rows: [],
+				}
+			},
+		},
+		element: {
+			type: Object,
+			default: null,
+		},
+		createMissingColumns: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	data() {
+		return {
+			editingColumnIndex: null,
+			editingColumnPreset: null,
+			existColumnOptions: [],
+			columnsConfig: [],
+		}
+	},
+
+	computed: {
+		...mapGetters(['isView']),
+	},
+
+	watch: {
+		columnsConfig: {
+			handler(newVal) {
+				this.$emit('update:columns', newVal)
+			},
+			deep: true,
+		},
+	},
+
+	async mounted() {
+		const existColumns = await this.$store.dispatch('getColumnsFromBE', {
+			tableId: !this.isView ? this.element.id : null,
+			viewId: this.isView ? this.element.id : null,
+		})
+		this.existColumnOptions = existColumns.map(column => ({
+			id: column.id,
+			label: column.title,
+		}))
+
+		for (let i = 0; i < this.previewData.columns.length; i++) {
+			const typeFull = this.previewData.columns[i].type + (this.previewData.columns[i].subtype ? `-${this.previewData.columns[i].subtype}` : '')
+			this.columnsConfig.push({
+				titleRaw: this.previewData.columns[i].title,
+				title: this.previewData.columns[i].title,
+				description: t('tables', 'This column was automatically created by the import service.'),
+				action: this.previewData.columns[i].id ? 'exist' : this.createMissingColumns ? 'new' : 'ignore',
+				existColumn: this.previewData.columns[i].id ? this.existColumnOptions.find(col => col.id === this.previewData.columns[i].id) : null,
+				type: this.previewData.columns[i].type,
+				subtype: this.previewData.columns[i].subtype,
+				typeFull,
+				typeLabel: this.getTypeLabel(typeFull),
+				numberDecimals: this.previewData.columns[i].numberDecimals,
+				numberPrefix: this.previewData.columns[i].numberPrefix,
+			})
+		}
+	},
+
+	methods: {
+		t,
+		editColumn(index) {
+			this.editingColumnPreset = this.columnsConfig[index]
+			this.editingColumnIndex = index
+			emit('tables:column:create', { isView: this.isView, element: this.element, onSave: this.onSaveColumn, preset: this.editingColumnPreset })
+
+			// Fix modal overlay
+			setTimeout(() => {
+				const modal = document.querySelector('.modal__content.create-column').closest('.modal-mask')
+				document.body.appendChild(modal)
+			}, 500)
```

Comment:
When user edits the new column format, create column modal will show up. But some time, it might be overlapped by import modal somehow. So, we should make sure the create column modal element is placed at the end of <body>

Reply:
Okay, thanks for explaining. How did you choose the `500` ms delay? Is it possible that 500ms might not be enough in some cases?

Reply:
The create column modal will show up instantly. There is no http request on that action. So, 500ms delay should be enough.